### PR TITLE
Feature/integration tests

### PR DESCRIPTION
**Pull request by M-Anwar-Hussaini**
# New features and changes in this milestone: 
- Solve the (N+1) problem for posts and comments
  - Before `includes` method:

  - image
  - After `inclues` method: 
  - image
- Add integration tests for the following views and controllers:
  - `user#index`
  - `user#show`
  - `post#index`
  - `post#show`

### CODE REVIEW

**Review by Qoosim — COMMENTED** [review]
#  *Status: Invalid Request* :red_circle: 

Hi @M-Anwar-Hussaini, 

While you made a great effort in this project, unfortunately, I cannot proceed to review your code.



## Invalid Code Review Request

You have submitted a project, that is a pair-programming but contains commits only from on partner. Kindly wait for the reply from student success and attach their email given to you to your PR if you're later allowed to work sole on this milestone.

Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

--------

_Invalid Code Review Request **does not count** into the code reviews limit._

**Review by Qoosim — COMMENTED** [review]
#  *Status: Invalid Request* :red_circle: 

Hi @team, 

While you made a great effort in this project, unfortunately, I cannot proceed to review your code.



## Invalid Code Review Request

You have submitted a project, that both partners do not have a fair contribution to the PR. Editing only the readme file is not sufficed to be considered as contributions. Every members must have a fair commit to the PR. 

Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

--------

_Invalid Code Review Request **does not count** into the code reviews limit._

File: README.md [hunk 1]
```diff
@@ -113,6 +112,11 @@ Open your web browser and go to http://localhost:3000 to access the Blog App.
...
 - LinkedIn: [Anwar Hussaini](https://www.linkedin.com/in/anwar-hussaini/)
 
+👤 **Gumaro Monroy Vazquez**
+
+- GitHub: [@Movgto](https://github.com/Movgto)
+- LinkedIn: [Gumaro Monroy Vazquez](https://www.linkedin.com/in/gumaro-monroy-vazquez-1705aa165/)
+
 ## 🤝 Contributing <a name="contributing"></a>
 
 Contributions, issues, and feature requests are welcome!
```
Comment (ghost) [inline README.md:122]: - [ ] 🌟 Great work on the documentation! However, there's a tiny hiccup 🚧 - the MIT License link seems to be broken. Let's ensure that there is a License in our repo and be sure to update the URL to keep that professionalism shining! ✨ Keep up the good work! 👏

File: spec/features/user_show_spec.rb [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+    @post3 = Post.create(title: 'Post 3', text: 'Post 3 text', author: @user, comments_counter: 0, likes_counter: 0)
+    visit user_path(@user)
+  end
+
+  scenario 'I can see the users profile picture.' do
+    profile = all('img')
+    expect(profile.size).to eq(1)
+  end
+
+  scenario 'I can see the users name.' do
```
Comment (ghost) [inline spec/features/user_show_spec.rb:18]: - [ ] In your `user#show` specs, Kindly add:
   - Tests that redirects to the post's show page when you click on a user's post.

File: spec/features/post_index_spec.rb [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+    @post2 = Post.create(title: 'Post 2', text: 'Post 2 text', author: @user, comments_counter: 0, likes_counter: 0)
+    @post3 = Post.create(title: 'Post 3', text: 'Post 3 text', author: @user, comments_counter: 0, likes_counter: 0)
+
+    Comment.create(user_id: @user.id, post_id: @post1.id, text: 'Comment 1')
+    Comment.create(user_id: @user.id, post_id: @post1.id, text: 'Comment 2')
+    Comment.create(user_id: @user.id, post_id: @post1.id, text: 'Comment 3')
+
+    Like.create(user_id: @user.id, post_id: @post1.id)
+    Like.create(user_id: @user.id, post_id: @post1.id)
+    Like.create(user_id: @user.id, post_id: @post1.id)
```
Comment (ghost) [inline spec/features/post_index_spec.rb:15]: - [ ] In your `post#index` specs, Kindly add:

     - Tests that asserts you can see how many comments a post has.
     - Tests that asserts you can see how many likes a post has.
     - Tests that asserts you can see a section for pagination if there are more posts than fit on the view.
     - Tests that redirects to the post's show page when you click on a post.

**Review by frank1738 — CHANGES_REQUESTED** [review]
Hi Team, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!



## Highlights

✅ Solved the `N+1` problem
✅ Tests covered are passing 👍 


## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

N/A
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: spec/features/post_index_spec.rb
Comment (LionRouge1) [inline spec/features/post_index_spec.rb:1]: - [ ] As mentioned by the previous reviewer there are some test cases that you have not covered. According to the project requirement we should have a test:
   - that asserts how many comments a post has.
   - that asserts  how many likes a post has.
   - that asserts a section for pagination if there are more posts than fit on the view.
 
 Kindly add those tests cases in your `post_index_spec.rb` file to follow the project requirement.
Screenshot from 2023-11-15 18-57-09

**Review by LionRouge1 — CHANGES_REQUESTED** [review]
Hi @M-Anwar-Hussaini, 

Great work on making the changes 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.

almost

## Suggested changes :recycle: 

_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by cynthiainga — APPROVED** [review]
Hi @M-Anwar-Hussaini, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



## Highlights

- [ ] All tests are passing ✔️ 
- [ ] All changes are made ✔️ 
- [ ] Correct Gitflow ✔️ 
- [ ] Has `Capybara` gem in `Gemfile` for integration test ✔️ 

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better. Some of them were missed by the previous reviewer and addressing them will really improve your application._

Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think the code review was unfair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**